Pod has no access

Straight after a brown field installation:
E0715 14:47:06.227604       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1beta1.Ingress: ingresses.extensions is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "ingresses" in API group "extensions" at the cluster scope

**To Reproduce**
Brown field installation

**Ingress Controller details**
Name:               agic-ingress-azure-749d44576b-l94vf
Namespace:          default
Priority:           0
PriorityClassName:  <none>
Node:               aks-agentpool-71662594-7/10.234.19.142
Start Time:         Mon, 15 Jul 2019 16:31:07 +0200
Labels:             aadpodidbinding=agic-ingress-azure
                    app=ingress-azure
                    pod-template-hash=749d44576b
                    release=agic
Annotations:        <none>
Status:             Running
IP:                 10.234.19.216
Controlled By:      ReplicaSet/agic-ingress-azure-749d44576b
Containers:
  ingress-azure:
    Container ID:   docker://d263ff28a4ba15145fd0e534d6f8e2dfa751af7fabd68a4b1c1eb0264e930b47
    Image:          mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:0.7.1
    Image ID:       docker-pullable://mcr.microsoft.com/azure-application-gateway/kubernetes-ingress@sha256:7474a032ed48a82105b960844e32fef203067a60e9a46a5c7a298b103c38725c
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Mon, 15 Jul 2019 16:31:09 +0200
    Ready:          True
    Restart Count:  0
    Environment Variables from:
      agic-cm-ingress-azure  ConfigMap  Optional: false
    Environment:
      KUBERNETES_PORT_443_TCP_ADDR:  aahnl-kubernetes-ot-78c6386c.hcp.westeurope.azmk8s.io
      KUBERNETES_PORT:               tcp://aahnl-kubernetes-ot-78c6386c.hcp.westeurope.azmk8s.io:443
      KUBERNETES_PORT_443_TCP:       tcp://aahnl-kubernetes-ot-78c6386c.hcp.westeurope.azmk8s.io:443
      KUBERNETES_SERVICE_HOST:       aahnl-kubernetes-ot-78c6386c.hcp.westeurope.azmk8s.io
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from agic-sa-ingress-azure-token-vsqhh (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  agic-sa-ingress-azure-token-vsqhh:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  agic-sa-ingress-azure-token-vsqhh
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type    Reason     Age   From                               Message
  ----    ------     ----  ----                               -------
  Normal  Scheduled  19m   default-scheduler                  Successfully assigned default/agic-ingress-azure-749d44576b-l94vf to aks-agentpool-71662594-7
  Normal  Pulling    19m   kubelet, aks-agentpool-71662594-7  pulling image "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:0.7.1"
  Normal  Pulled     19m   kubelet, aks-agentpool-71662594-7  Successfully pulled image "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:0.7.1"
  Normal  Created    19m   kubelet, aks-agentpool-71662594-7  Created container
  Normal  Started    19m   kubelet, aks-agentpool-71662594-7  Started container

* Output of `kubectl logs <ingress controller>. 
E0715 14:50:36.835413       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "secrets" in API group "" at the cluster scope
E0715 14:50:36.994211       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1beta1.Ingress: ingresses.extensions is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "ingresses" in API group "extensions" at the cluster scope
E0715 14:50:36.994220       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1.Pod: pods is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "pods" in API group "" at the cluster scope
E0715 14:50:36.994255       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1.Endpoints: endpoints is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "endpoints" in API group "" at the cluster scope
E0715 14:50:36.996109       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1.Service: services is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "services" in API group "" at the cluster scope
E0715 14:50:37.838553       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "secrets" in API group "" at the cluster scope
E0715 14:50:38.000190       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1.Endpoints: endpoints is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "endpoints" in API group "" at the cluster scope
E0715 14:50:38.000198       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1.Pod: pods is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "pods" in API group "" at the cluster scope
E0715 14:50:38.000922       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1beta1.Ingress: ingresses.extensions is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "ingresses" in API group "extensions" at the cluster scope
E0715 14:50:38.001277       1 reflector.go:125] github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext/context.go:296: Failed to list *v1.Service: services is forbidden: User "system:serviceaccount:default:agic-sa-ingress-azure" cannot list resource "services" in API group "" at the cluster scope

* Any Azure support tickets associated with this issue.
???

robertka@OT-0038:~$ helm list
NAME                            REVISION        UPDATED                         STATUS          CHART                           APP VERSION     NAMESPACE
agic                            4               Tue Jul 16 08:37:46 2019        DEPLOYED        ingress-azure-0.7.1             0.7.1           default

The service and its port definitely exist:
robertka@OT-0038:~$ kubectl -n development get svc ssldemo-svc -o yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"name":"ssldemo-svc","namespace":"development"},"spec":{"ports":[{"name":"http-ssldemo","port":80,"targetPort":"http-main"}],"selector":{"app":"ssldemo"},"type":"ClusterIP"}}
  creationTimestamp: "2019-05-13T09:47:06Z"
  name: ssldemo-svc
  namespace: development
  resourceVersion: "5463974"
  selfLink: /api/v1/namespaces/development/services/ssldemo-svc
  uid: 11da7b15-7564-11e9-88f6-fedc648087ef
spec:
  clusterIP: 10.234.7.27
  ports:
  - name: http-ssldemo
    port: 80
    protocol: TCP
    targetPort: http-main
  selector:
    app: ssldemo
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}

Also, the pods for label 'app:ssldemo' have this port specification:
ports:
      - containerPort: 8080
        name: http-main
        protocol: TCP

After deleting the ingress controller pod and viewing the logs from the new pod directly after startup, I notice this:
robertka@OT-0038:/mnt/d/src/ssl$ kubectl logs agic-ingress-azure-749d44576b-bpvf8 --tail=1000 -f
ERROR: logging before flag.Parse: I0717 07:49:46.160366       1 main.go:233] Using verbosity level 3 from environment variable APPGW_VERBOSITY_LEVEL
I0717 07:49:46.160473       1 main.go:168] Creating authorizer from Azure Managed Service Identity
E0717 07:49:46.200235       1 main.go:179] Error getting Application Gatewayag-otazure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/7b520150-0df3-4adb-83b7-5646c7c62fef/resourceGroups/network-ot-rg/providers/Microsoft.Network/applicationGateways/ag-ot?api-version=2018-12-01: StatusCode=403 -- Original Error: adal: Refresh request failed. Status Code = '403'. Response body: **no AzureAssignedIdentity found for pod:default/agic-ingress-azure-749d44576b-bpvf8**
I0717 07:49:46.200261       1 main.go:180] Retrying in 10s
I0717 07:49:56.201593       1 main.go:109] Ingress Controller will observe all namespaces.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x12b89ca]

goroutine 1 [running]:
github.com/Azure/application-gateway-kubernetes-ingress/pkg/appgw.validateFrontendIPConfiguration(0x185e4e0, 0xc00042b800, 0x0, 0xc000048016, 0x24, 0xc000042045, 0xd, 0xc00004600b, 0x5, 0x0, ...)
        /gopath/src/github.com/Azure/application-gateway-kubernetes-ingress/pkg/appgw/validators.go:119 +0x3a
github.com/Azure/application-gateway-kubernetes-ingress/pkg/appgw.FatalValidateOnExistingConfig(0x185e4e0, 0xc00042b800, 0x0, 0xc000048016, 0x24, 0xc000042045, 0xd, 0xc00004600b, 0x5, 0x0, ...)
        /gopath/src/github.com/Azure/application-gateway-kubernetes-ingress/pkg/appgw/validators.go:153 +0xc0
main.main()
        /gopath/src/github.com/Azure/application-gateway-kubernetes-ingress/cmd/appgw-ingress/main.go:121 +0x83c
(pod keeps restarting)

I did install aadPodIdentity, but I haven't used it for other solutions. Did I perhaps miss a step to configure it or should the helm installation take care of that?

--- a/pkg/appgw/appgw_test.go
+++ b/pkg/appgw/appgw_test.go
@@ -122,7 +122,7 @@ var _ = Describe("Tests `appgw.ConfigBuilder`", func() {
 		Spec: v1.ServiceSpec{
 			Ports: []v1.ServicePort{
 				{
-					Name: "frontendPort",
+					Name: "servicePort",
 					TargetPort: intstr.IntOrString{
 						Type:   intstr.String,
 						StrVal: backendName,
@@ -163,7 +163,7 @@ var _ = Describe("Tests `appgw.ConfigBuilder`", func() {
 				},
 				Ports: []v1.EndpointPort{
 					{
-						Name:     backendName,
+						Name:     "servicePort",
 						Port:     backendPort,
 						Protocol: v1.ProtocolTCP,
 					},

--- a/pkg/appgw/backendhttpsettings.go
+++ b/pkg/appgw/backendhttpsettings.go
@@ -177,9 +177,8 @@ func (c *appGwConfigBuilder) getBackendsAndSettingsMap(cbCtx *ConfigBuilderConte
 							resolvedBackendPorts[pair] = nil
 						} else {
 							// if service port is defined by name, need to resolve
-							targetPortName := sp.TargetPort.StrVal
-							glog.V(1).Infof("resolving port name %s", targetPortName)
-							targetPortsResolved := c.resolvePortName(targetPortName, &backendID)
+							glog.V(3).Infof("resolving port name %s", sp.Name)
+							targetPortsResolved := c.resolvePortName(sp.Name, &backendID)
 							for targetPort := range targetPortsResolved {
 								pair := serviceBackendPortPair{
 									ServicePort: sp.Port,

--- a/pkg/appgw/health_probes.go
+++ b/pkg/appgw/health_probes.go
@@ -7,12 +7,13 @@ package appgw
 
 import (
 	"fmt"
+	"sort"
+
 	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
 	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
-	"sort"
 
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/annotations"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/brownfield"
@@ -124,7 +125,7 @@ func (c *appGwConfigBuilder) getProbeForServiceContainer(service *v1.Service, ba
 				// port is defined as port number
 				allPorts[sp.TargetPort.IntVal] = nil
 			} else {
-				for targetPort := range c.resolvePortName(sp.TargetPort.StrVal, &backendID) {
+				for targetPort := range c.resolvePortName(sp.Name, &backendID) {
 					allPorts[targetPort] = nil
 				}
 			}

